compiler/src/dotty/tools/dotc/transform/Inlining.scala

```diff
@@ -59,8 +63,16 @@ class Inlining extends MacroTransform {
   private class InliningTreeMap extends TreeMapWithImplicits {
     override def transform(tree: Tree)(using Context): Tree = {
       tree match
-        case tree: DefTree =>
+        case tree: MemberDef =>
           if tree.symbol.is(Inline) then tree
+          else if tree.symbol.is(Param) then super.transform(tree)
+          else if
+            !tree.symbol.isPrimaryConstructor
+            && StagingContext.level == 0
+            && TastyAnnotations.hasTastyAnnotation(tree.symbol)
+          then
+            val trees = new TastyAnnotations(Inlining.this).expandAnnotations(tree)
```

Comment:
Style: We usually prefer an explicit self name like `thisPhase` over a qualified `Inlining.this`.

Reply:
Updated